### PR TITLE
Remove minitest gem altogether, make specs more rspec-y, add a spec for the array form of :on

### PR DESCRIPTION
**Pull request by apurvis**
Sorry; didn't realize I forgot to remove `minitest` on the previous PR.  PR also includes a reorganization of specs to make them more in the rspec style.

### CODE REVIEW

**Comment by apurvis** [comment]
i did a larger scale refactor of the specs to make things more "rspec style".  no functional changes in any of the specs.

here's the current pretty print output of `bundle exec rspec --format documentation`:

```
Retriable::Config
  raises errors on invalid configuration
  defaults
    sleep defaults to enabled
    tries defaults to 3
    max interval defaults to 60
    randomization factor defaults to 0.5
    multiplier defaults to 1.5
    max elapsed time defaults to 900
    intervals defaults to nil
    timeout defaults to nil
    on defaults to [StandardError]
    on retry handler defaults to nil
    contexts defaults to {}

Retriable::ExponentialBackoff
  generates 10 randomized intervals
  generates defined number of intervals
  generates intervals with a defined base interval
  generates intervals with a defined multiplier
  generates intervals with a defined max interval
  generates intervals with a defined rand_factor
  generates 10 non-randomized intervals
  defaults
    tries defaults to 3
    max interval defaults to 60
    randomization factor defaults to 0.5
    multiplier defaults to 1.5

Retriable
  global scope extension
    cannot be called in the global scope without requiring the core_ext/kernel
    can be called once the kernel extension is required
  #retriable
    raises a LocalJumpError if not given a block
    stops at first try if the block does not raise an exception
    makes 3 tries when retrying block of code raising StandardError with no arguments
    makes only 1 try when exception raised is not ancestor of StandardError
    with custom exception tries 3 times and re-raises the exception
    tries 10 times
    will timeout after 1 second
    applies a randomized exponential backoff to each try
    with custom defined intervals
    runs for a max elapsed time of 2 seconds
    raises ArgumentError on invalid options
    with rand_factor 0.0
      obeys a max interval of 1.5 seconds
      with an on_retry handler, 6 max retries
        applies a non-randomized exponential backoff to each try
    with a hash exception list
      where the value is an exception message pattern
      matches exception subclasses
      does not retry matching exception subclass but not message
      successfully retries when the values are arrays of exception message patterns
  #configure
    raises NoMethodError on invalid configuration
  #with_context
    stops at first try if the block does not raise an exception
    respects the context options
    allows override options
    raises an ArgumentError when the context isn't found

Finished in 4.03 seconds (files took 0.41565 seconds to load)
47 examples, 0 failures
```

File: spec/retriable_spec.rb [hunk 1]
```diff
@@ -207,178 +172,131 @@
...
 
       expect do
-        retriable do
-          @tries += 1
-          raise StandardError
+        described_class.retriable(
+          base_interval: 1.0,
+          multiplier: 1.0,
+          rand_factor: 0.0,
+          max_elapsed_time: 2.0
```
[the remark this answers is not in the document]
Comment (apurvis) [reply spec/retriable_spec.rb]: @kamui is this really a style we want to enforce?  i'm happy to change it but i've always thought trailing commas were kind of undesirable.

**Comment by apurvis** [comment]
@kamui any questions/thoughts on this PR, style or otherwise?  it only touches the specs and just makes them "rspec style".

**Comment by olleolleolle** [comment]
This is supernice.

**Comment by kamui** [comment]
Thanks for the PR!

**Pull request merged**